src/main/java/com/cgdp/library/controller/AuthorController.java

```diff
@@ -0,0 +1,35 @@
+package com.cgdp.library.controller;
+
+import com.cgdp.library.dto.author.CreateAuthorResponseDTO;
+import com.cgdp.library.dto.author.CreateAuthorRequestDTO;
+import com.cgdp.library.entity.AuthorEntity;
+import com.cgdp.library.service.AuthorService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@AllArgsConstructor
+@RequestMapping("/authors")
+@RestController
+public class AuthorController {
+
+    private final AuthorService authorService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public CreateAuthorResponseDTO createAuthor(@RequestBody CreateAuthorRequestDTO createAuthorRequestDTO) {
+
+        String authorName = createAuthorRequestDTO.name();
+        AuthorEntity createdAuthor = authorService.save(authorName);
```

Comment:
AuthorEntity is part of the persistence layer (repository) it shouldn't reach this far
Let's do the following, make the service layer return an AuthorDTO
```suggestion
        AuthorDTO newAuthor = authorService.save(authorName);
```